core/domain/suggestion_services.py

```diff
@@ -171,17 +194,37 @@ def _update_suggestion(suggestion):
     Args:
         suggestion: Suggestion. The suggestion to be updated.
     """
-    suggestion.validate()
 
-    suggestion_model = suggestion_models.GeneralSuggestionModel.get_by_id(
-        suggestion.suggestion_id)
+    _update_suggestions([suggestion])
+
+
+def _update_suggestions(suggestions):
+    """Updates the given suggestions.
+
+    Args:
+        suggestions: list(Suggestion). The suggestions to be updated.
+    """
+    suggestion_ids = []
 
-    suggestion_model.status = suggestion.status
-    suggestion_model.final_reviewer_id = suggestion.final_reviewer_id
-    suggestion_model.change_cmd = suggestion.change.to_dict()
-    suggestion_model.score_category = suggestion.score_category
+    for suggestion in suggestions:
+        suggestion.validate()
+        suggestion_ids.append(suggestion.suggestion_id)
 
-    suggestion_model.put()
+    suggestion_models_to_update = (
+        suggestion_models.GeneralSuggestionModel.get_multi(suggestion_ids)
+    )
+
+    for suggestion_model in suggestion_models_to_update:
+        suggestion = python_utils.NEXT(
```

Comment:
Why not do an enumerate() in the for loop and fetch directly by the index? The result of the get_multi() should match the order of `suggestion_ids` which matches the order of suggestions`.